Tools/package_server_build.py

```diff
@@ -0,0 +1,288 @@
+]
+
+def main() -> None:
+    parser = argparse.ArgumentParser(
+        description="Packages the SS14 content repo for release on all platforms.")
+    parser.add_argument("--platform",
+                        "-p",
+                        action="store",
+                        choices=PLATFORM_RIDS,
+                        nargs="*",
+                        help="Which platform to build for. If not provided, all platforms will be built")
+
+    parser.add_argument("--skip-build",
+                        action="store_true",
+                        help=argparse.SUPPRESS)
+
+    parser.add_argument("--hybrid-acz",
+                        action="store_true",
+                        help="Creates a 'Hybrid ACZ' build that contains an embedded Content.Client.zip the server hosts.")
+
+    args = parser.parse_args()
+    platforms = args.platform
+    skip_build = args.skip_build
+    hybrid_acz = args.hybrid_acz
+
+    if not platforms:
+        platforms = PLATFORM_RIDS_DEFAULT
+
+    if os.path.exists("release"):
+        print(Fore.BLUE + Style.DIM +
+              "Cleaning old release packages (release/)..." + Style.RESET_ALL)
+        shutil.rmtree("release")
+
+    os.mkdir("release")
+
+    if hybrid_acz:
+        # Hybrid ACZ involves a file "Content.Client.zip" in the server executable directory.
+        # Rather than hosting the client ZIP on the watchdog or on a separate server,
+        #  Hybrid ACZ uses the ACZ hosting functionality to host it as part of the status host,
+        #  which means that features such as automatic UPnP forwarding still work properly.
+        import package_client_build
+        package_client_build.build(skip_build)
+
+    # Good variable naming right here.
+    for platform in PLATFORMS:
+        if platform.rid not in platforms:
+            continue
+
+        build_platform(platform, skip_build, hybrid_acz)
+
+def wipe_bin():
+    print(Fore.BLUE + Style.DIM +
+          "Clearing old build artifacts (if any)..." + Style.RESET_ALL)
+    if os.path.exists(p("RobustToolbox", "bin")):
+        shutil.rmtree(p("RobustToolbox", "bin"))
+
+    if os.path.exists("bin"):
+        shutil.rmtree("bin")
+
+
+def build_platform(platform: PlatformReg, skip_build: bool, hybrid_acz: bool) -> None:
+    print(Fore.GREEN + f"Building project for {platform.rid}..." + Style.RESET_ALL)
+
+    if not skip_build:
+        subprocess.run([
+            "dotnet",
+            "build",
+            p("OpenDreamServer", "OpenDreamServer.csproj"),
+            "-c", "Release",
+            "--nologo",
+            "/v:m",
+            f"/p:TargetOS={platform.target_os}",
+            "/t:Rebuild",
+            "/p:FullRelease=True",
+            "/m"
+        ], check=True)
+
+        publish_client_server(platform.rid, platform.target_os)
+
+    print(Fore.GREEN + "Packaging {platform.rid} server..." + Style.RESET_ALL)
+    server_zip = zipfile.ZipFile(p("release", f"SS14.Server_{platform.rid}.zip"), "w",
```

Comment:
Would changing the name of this zip break anything?

Reply:
See other reply.